mrn_mysql_compat.h

```diff
@@ -962,3 +962,31 @@ typedef uint mrn_srid;
 #  define MRN_SET_MESSAGE_FROM_CTX(ctx, error_code)     \
   my_message(error_code, ctx->errbuf, MYF(0))
 #endif
+
+#if defined(MRN_MARIADB_P) &&                                      \
+    ((MYSQL_VERSION_ID >= 100526 && MYSQL_VERSION_ID < 100600) ||  \
```

Comment:
`MYSQL_VERSION_ID < 100600`
Do we need an upper limit requirement?

Reply:
I think we need it because MariaDB v 11.4 would be matched if we didn't set the upper limit.

Reply:
Yes.
an upper limit is need.

For example, we should use `(table_list->get_db_name())` in MariaDB v10.6.18. 
Because MariaDB v10.6.18 has not `table_list->get_db_name().str`.
So , we fail to build Mroonga if we also use `(table_list->get_db_name().str)` in MariaDB v10.6.18.

Reply:
I get it.